docs/custom_component.md

```diff
@@ -0,0 +1,81 @@
+# Custom component
+
+The goal of Fondant is to make it easy for people to define new components which can be incorporated into data preparation pipelines. This allows to share and reuse them as building blocks when preparing data for Foundation models.
+
+A Fondant component should implement one logical piece of data preprocessing, like captioning images or removing PPI from text. It's important to not include too much logic into a single component, as it might be beneficial to split up the logic into 2 separate components in that case.
```

Comment:
I had to look up `PPI` but I think it should be `PII` either way I would write it out?